interp/deprecation.ml

```diff
@@ -12,10 +12,11 @@ type t = { since : string option ; note : string option }
 
 let make ?since ?note () = { since ; note }
 
-let create_warning ~object_name ~warning_name name_printer =
+let create_warning ~object_name ~warning_name name_printer ?loc (qid,depr) =
```

Comment:
Shouldn't the argument name be changed to clarify its purpose? At the very least, the documentation in the `.mli` should be updated.

```suggestion
let create_warning ~object_name ~warning_name_if_no_since name_printer ?loc (qid,depr) =
```